docs/pattern-matching.md

```diff
@@ -123,18 +132,19 @@ Parse = λarg0 switch matched = (- arg0 '(') {
   }
 }
 ```
-Unlike with `switch`, with pattern matching equations you can't access the value of the predecessor of the matched value directly, but instead you can match on a variable.
+A key difference from direct switch expressions is how variable binding works. In pattern matching equations, you can't directly access predecessor values. Instead, variables (like n above) are bound to computed expressions based on the matched value.
 Notice how in the example above, `n` is bound to `(+ 1 matched-1)`.
 
 Notice that this definition is valid, since `*` will cover both `p` and `0` cases when the first argument is `False`.
 
+This example shows how pattern order matters, with wildcards covering multiple specific cases:
```

Comment:
Maybe you should instead say something like "patterns are considered from top to bottom". Also, i think it would fit better in the same paragraph as the sentence above